### PR TITLE
BUG: Missing cuda keys

### PR DESCRIPTION
**Pull request by carterbox**
The keys `cuda_compiler` and `cuda_compiler_version_min` were added to the conda-forge pinnings, but they are not added here in staged recipes. Without them, CUDA build jobs will try to install the `none_linux-64` package instead of `nvcc_linux-64`. Also added `cuda_compiler_version_min` to be consistent with the pinnings, and since I don't know how it is used.

EDIT: I now realize that the keys in staged recipes override the ones from the pinnings, so the error is due to the overrided keys here disrupting the key zipping from the pinnings. Thus, I should include all cuda related keys that zip with `cuda_compiler_verion`.

~~Also added a line to the build steps to squash an error from running the build-locally script.~~
EDIT: Removed docker permissions related changes

Related to: https://github.com/conda-forge/staged-recipes/pull/22898 , https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4407

### CODE REVIEW

File: .ci_support/linux64_cuda111.yaml [hunk 1]
```diff
@@ -22,3 +22,7 @@ docker_image:
 - quay.io/condaforge/linux-anvil-cuda:11.1
 cuda_compiler_version:
 - 11.1
+cuda_compiler:
+- nvcc
+cuda_compiler_version_min:
+- 10.2
```
Comment (jakirkham) [inline .ci_support/linux64_cuda111.yaml:28]:
```suggestion
- 10.2
cudnn:
- 8
```

Comment (carterbox) [reply .ci_support/linux64_cuda111.yaml:28]: Thanks!

File: .scripts/build_steps.sh [hunk 2]
```diff
@@ -38,6 +38,7 @@ pushd /home/conda/staged-recipes/recipes > /dev/null
 if [ "${AZURE}" == "True" ]; then
     git fetch --force origin main:main
 fi
+git config --global --add safe.directory /home/conda/staged-recipes
```
Comment (jakirkham) [inline .scripts/build_steps.sh:41]: Why is this needed?

Comment (carterbox) [reply .scripts/build_steps.sh:41]:
```
+ echo 'Finding recipes merged in main and removing them from the build.'
Finding recipes merged in main and removing them from the build.
+ pushd /home/conda/staged-recipes/recipes
+ '[' False == True ']'
+ git ls-tree --name-only main -- .
+ xargs -I '{}' sh -c 'rm -rf ~/staged-recipes-copy/recipes/{} && echo Removing recipe: {}'
fatal: detected dubious ownership in repository at '/home/conda/staged-recipes'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/conda/staged-recipes
Traceback (most recent call last):
  File "/home/dching/Documents/staged-recipes/build-locally.py", line 101, in <module>
    main()
  File "/home/dching/Documents/staged-recipes/build-locally.py", line 95, in main
    run_docker_build(ns)
  File "/home/dching/Documents/staged-recipes/build-locally.py", line 33, in run_docker_build
    subprocess.check_call([script])
  File "/usr/lib64/python3.11/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['.scripts/run_docker_build.sh']' returned non-zero exit status 1.
```
I get the following error message when using the `build-locally.py`

Comment (jakirkham) [reply .scripts/build_steps.sh:41]: Thanks for the clarification 🙏

Could we please split this out as a separate issue or PR?

It seems unrelated to the changes here (and we may want to evaluate our options for solving it including whether the Docker images themselves need changes)

Comment (carterbox) [reply .scripts/build_steps.sh:41]: Yes

**Comment by jakirkham** [comment]
Thanks Daniel! 🙏

File: .scripts/build_steps.sh (hunk 2 above)
Comment (jakirkham) [reply .scripts/build_steps.sh]: Sounds good. Please feel free to link an issue here (once opened). Agree we should figure out some kind of solution. Think it may even be needed in feedstocks

Comment (vicentebolea) [reply .scripts/build_steps.sh]: How did you guys get around to this issue I am hitting the same thing when using the build_locally.py script

Comment (carterbox) [reply .scripts/build_steps.sh]: Add the topic line of this conversation to the build_steps script.

Comment (vicentebolea) [reply .scripts/build_steps.sh]: What I wonder is that this change cannot make it to the feedstock project, thus I would have to add it anytime I want to build locally, would there be an alternative?

Comment (jakirkham) [reply .scripts/build_steps.sh]: If the feedstock hasn't been updated in a while, please make sure to [re-render]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#dev-update-rerender )
